src/lib/provable/gadgets/elliptic-curve.ts

```diff
@@ -281,6 +282,9 @@ function verifyEcdsaGeneric(
   // if we allowed non-canonical Rx, the prover could make verify() return false on a valid signature, by adding a multiple of `Curve.order` to Rx.
   ForeignField.assertLessThan(Rx, Curve.order);
 
+  // assert s to be canonical
+  if (checkSCanonical) ForeignField.assertLessThan(s, Curve.order);
```

Comment:
I decided to only assert `s` to be canonical and leave `r` as is. Since `r` will be checked to be canonical but just returns `false` instead of throwing an error, I thought this would be the least breaking case and save constraints

Reply:
makes sense!